steps/container.py

```diff
@@ -233,3 +185,71 @@ def get_output(self, history=True):
     def remove_image(self, force=False):
         self.logging.info("Removing image %s" % self.image_id)
         d.remove_image(image=self.image_id, force=force)
+
+    def copy_file_to_container(self, src_file, dest_folder):
+        full_src_file=os.path.join(os.getcwd(), src_file)
+
+        # The Docker library needs tar bytes to put_archive
+        f = tempfile.NamedTemporaryFile()
+        t = tarfile.open(mode='w', fileobj=f)
```

Comment:
It is much better to use the `with` statement here: https://realpython.com/read-write-files-python/#opening-and-closing-a-file-in-python It will be shorter and less error-prone